### DISCUSSION TITLE
Why isn't the logo transparent enough? The opacity key seems not working.

### DISCUSSION THREAD

**Discussion by skyleaworlder** [TJ-CSCCG/tongji-recommendation-template#1] [Q&A] [answered]
为什么 pdf 版本的校徽水印不透明？

---

Original Title: How about replace logo.svg with logo.pdf?

如果我 Windows texlive 没装错的话，svg 在 Windows texlive-2019 这边应该不是默认宏包。

Windows 这边对于 texlive 的安装、更新的支持要整体弱于 Unix。所以我考虑为了使用方便是否可以用 pdf？

注：只是为了讨论，想看看 @tobiasforest 的想法。

**Comment by RizhongLin** [comment]
我的 windows texlive 2022 里有 svg 宏包。

**Reply by skyleaworlder** [reply]
抱歉上面提供了一个错误信息，windows 2019 确实装了 svg，但是 2019 的 svg 存在 bug，同样的问题被发现在：https://gitlab.com/inkscape/inkscape/-/issues/517 与 https://github.com/mrpiggi/svg/issues/13。

**Reply by skyleaworlder** [reply]
XeTeX Version 3.141592653-2.6-0.999995 (TeX Live 2023)，0ls 有什么头绪吗

```text
Package svg Info: Calling Inkscape on input line 143.
runsystem(inkscape "logo.svg" -D --export-latex --export-filename="logo_svg-tex.pdf")...executed.

runsystem(inkscape "logo.svg" -D --export-latex \relax --without-gui --export-pdf="logo_svg-tex.pdf")...executed.


d:/TJ-CSCCG/tongji-recommendation-template/main.tex:143: Package svg Error: Inkscape version not detected.
```

**Reply by skyleaworlder** [reply]
补充一些信息：

```text
package:     svg-inkscape
category:    Package
shortdesc:   How to include an SVG image in LaTeX using Inkscape
longdesc:    The document demonstrates the use of SVG images in LaTeX documents. Using the "PDF+LaTeX output" option of Inkscape, it is possible to include SVG in documents, in which LaTeX typesets the text. This results in uniform text style throughout the document, including text in images; moreover, LaTeX commands may be used in the image's text, providing such things as mathematics and references. The document also describes how to automate the conversion from SVG to PDF+LaTeX using Inkscape.
installed:   Yes
revision:    32199
sizes:       doc: 361k
relocatable: No
cat-license: lppl1.3
cat-topics:  graphics-prep
cat-related: svg
collection:  collection-langenglish

package:     svg
category:    Package
shortdesc:   Include and extract SVG pictures in LaTeX documents
longdesc:    This bundle contains the two packages svg and svg-extract. The svg package is intended for the automated integration of SVG graphics into LaTeX documents. Therefore the capabilities provided by Inkscape -- or more precisely its command line tool -- are used to export the text within an SVG graphic to a separate file, which is then rendered by LaTeX. For this purpose the two commands \includesvg and \includeinkscape are provided which are very similar to the \includegraphics command of the graphicx package. In addition, the package svg-extract allows the extraction of these graphics into independent files in different graphic formats, exactly as it is rendered within the LaTeX document, using either ImageMagick or Ghostscript.
installed:   Yes
revision:    57010
sizes:       src: 221k, doc: 1757k, run: 85k
relocatable: No
cat-version: 2.02k
cat-license: lppl1.3c
cat-topics:  graphics graphics-incl graphics-import
cat-related: svg-inkscape
cat-contact-bugs: https://github.com/mrpiggi/svg/issues
cat-contact-repository: https://github.com/mrpiggi/svg
collection:  collection-latexextra
```

这是因为 `svg-inkscape` 没有 `cat-version` 导致的吗

**Reply by RizhongLin** [reply]
Tl;dr. I have made some modifications. See 28154b639f0d443f21392c03e030500ea41105f9

**Reply by skyleaworlder** [reply]
lol

**Reply by skyleaworlder** [reply]
Currently:

![image](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/45269589/946b3147-ee62-4c8d-b353-9d60bbdc93da)

😭

**Reply by skyleaworlder** [reply]
Key opacity seems not working at including a pdf.

**Reply by RizhongLin** [reply]
It appears that the issue might not be present on my Windows computer. I successfully compiled the project on Windows 11 using XeLaTeX, and it also works flawlessly on Overleaf. 

It is possible that the problem stems from either your PDF viewer, your TeXLive version, or your Windows version, but I cannot be entirely certain. I recommend checking these components to troubleshoot the problem.

Best,

**Reply by skyleaworlder** [reply]
Opacity failed with Windows 10 & XeTeX (TeX Live 2023) & {edge,chrome} pdf viewer.

**Reply by RizhongLin** [reply]
I cannot reproduce.

**Reply by RizhongLin** [reply]
目前我怀疑是 logo.pdf 的问题。不想再改了，README 里的使用方法目前只有 Overleaf。起码 Overleaf 上是正常的。

**Reply by RizhongLin** [reply]
你机器上有 Adobe Acrobat 吗？

**Reply by skyleaworlder** [reply]
> 目前我怀疑是 logo.pdf 的问题。不想再改了，README 里的使用方法目前只有 Overleaf。起码 Overleaf 上是正常的。

是的，我本来正在尝试增加推荐信模板的使用方式。

**Reply by skyleaworlder** [reply]
> 你机器上有 Adobe Acrobat 吗？

我正在尝试安装，不过我怀疑不是 pdf viewer 的问题。

---

确实不是

![image](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/45269589/1a14b8f4-48c9-474f-a33d-754e8bdce528)

**Reply by RizhongLin** [reply]
Acrobat 的 Tools -> Print Production -> Preflight -> List Transparent Objects 

![](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/59563695/a2b8fdcf-d0a9-4cd3-a933-6e5f60a54e25)

可以看到 alpha 值被设置为 0.1。

如果你的 main.pdf 这里也是 0.1 的话，那可以考虑不是 LaTeX 的问题……

**Reply by skyleaworlder** [reply]
![image](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/45269589/8293cc1e-e5e8-487e-a60d-a748b26ac69f)

确实 opacity 仍被置为 1.0，我可以确认我没有修改模板代码。我的编译指令是：

```shell
latexmk main.tex -halt-on-error -xelatex -shell-escape
```

**Reply by RizhongLin** [reply]
我现在复现了问题，发现第一页的水印不是透明的，但从第二页开始透明度设置就有效了。你也是这样吗？

LuaLaTeX 貌似没有问题。

**Reply by RizhongLin** [reply]
OK 现在 XeLaTeX 编译的版本每页水印都不透明了……

**Reply by skyleaworlder** [reply]
![image](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/45269589/50577414-9ce1-4536-8661-6823f68e94cc)

令我惊讶的是我并没有在 “preflight” 中看到第 2 页的透明度对象被识别。

并且我的第一页与第二页都不是透明的。

![image](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/45269589/e171d267-6dc7-46e4-9276-277e789f73ce)

**Reply by RizhongLin** [reply]
嗯嗯，我查了下应该是 XeLaTeX 不太支持 includegraphics 的 opacity 命令。那要么就把 logo.pdf 转成 jpeg，要么就只用 LuaLaTeX 编译。

**Reply by skyleaworlder** [reply]
😭 LuaLaTeX 编译在水印 opacity 这方面没有问题：

![image](https://github.com/TJ-CSCCG/tongji-recommendation-template/assets/45269589/6a580793-94ee-447e-82cc-37522d893f28)

**Reply by RizhongLin** [reply, marked as answer]
找到一条警告：

> Package epstopdf Warning: Drivers other than \`pdftex\' and \`luatex\' are not supported.

**Reply by skyleaworlder** [reply]
Nice find! 我这边也输出了相同的警告。

那我觉得这个问题算是被解决了，只要告知用户不要使用 XeTeX 就可以了（

**Comment by RizhongLin** [comment]
Yes!

**Comment by skyleaworlder** [comment]
PR Related: #2
